aidial_adapter_bedrock/llm/model/claude/v3/adapter.py

```diff
@@ -0,0 +1,161 @@
+from typing import List, Mapping, Optional, TypedDict, Union
+
+from aidial_sdk.chat_completion import Message
+from anthropic import NOT_GIVEN, NotGiven
+from anthropic.lib.bedrock import AsyncAnthropicBedrock
+from anthropic.lib.streaming import AsyncMessageStream
+from anthropic.types import (
+    MessageDeltaEvent,
+    MessageParam,
+    MessageStartEvent,
+    MessageStreamEvent,
+)
+
+from aidial_adapter_bedrock.dial_api.request import ModelParameters
+from aidial_adapter_bedrock.dial_api.storage import (
+    FileStorage,
+    create_file_storage,
+)
+from aidial_adapter_bedrock.dial_api.token_usage import TokenUsage
+from aidial_adapter_bedrock.llm.chat_model import ChatCompletionAdapter
+from aidial_adapter_bedrock.llm.consumer import Consumer
+from aidial_adapter_bedrock.llm.exceptions import ValidationError
+from aidial_adapter_bedrock.llm.model.claude.v3.converters import (
+    ClaudeFinishReason,
+    to_claude_messages,
+    to_dial_finish_reason,
+)
+from aidial_adapter_bedrock.llm.model.conf import DEFAULT_MAX_TOKENS_ANTHROPIC
+from aidial_adapter_bedrock.llm.tools.claude_emulator import (
+    legacy_tools_emulator,
+)
+from aidial_adapter_bedrock.utils.log_config import bedrock_logger as log
+
+
+class UsageEventHandler(AsyncMessageStream):
+    prompt_tokens: int = 0
+    completion_tokens: int = 0
+    stop_reason: Optional[ClaudeFinishReason] = None
+
+    async def on_stream_event(self, event: MessageStreamEvent):
+        if isinstance(event, MessageStartEvent):
+            self.prompt_tokens = event.message.usage.input_tokens
+        elif isinstance(event, MessageDeltaEvent):
+            self.completion_tokens += event.usage.output_tokens
+            self.stop_reason = event.delta.stop_reason
+
+
+class ChatParams(TypedDict):
+    max_tokens: int
+    stop_sequences: Union[List[str], NotGiven]
+    system: Union[str, NotGiven]
+    temperature: Union[float, NotGiven]
+    top_p: Union[float, NotGiven]
+
+
+class Adapter(ChatCompletionAdapter):
+    model: str
+    storage: Optional[FileStorage]
+    client: AsyncAnthropicBedrock
+
+    async def chat(
+        self,
+        consumer: Consumer,
+        params: ModelParameters,
+        messages: List[Message],
+    ):
+        if len(messages) == 0:
+            raise ValidationError("List of messages must not be empty")
+
+        tools_emulator = self.tools_emulator(params.tool_config)
+        base_messages = tools_emulator.parse_dial_messages(messages)
+        tool_stop_sequences = tools_emulator.get_stop_sequences()
+
+        prompt, claude_messages = await to_claude_messages(
+            base_messages, self.storage
+        )
+
+        completion_params = ChatParams(
+            max_tokens=params.max_tokens or DEFAULT_MAX_TOKENS_ANTHROPIC,
```

Comment:
Should 0 be considered as None?

Reply:
It couldn't be zero, because we have a check in SDK that max_tokens must be positive integer.
Turning 0 into None is quite suprising anyway.